tests/src/unit-deserialization.cpp

```diff
@@ -588,8 +588,7 @@ TEST_CASE("deserialization")
                 auto first = str.begin();
                 auto last = str.end();
                 json j;
-                auto ia = nlohmann::detail::input_adapter(str);
-                json_sax_dom_parser<json, decltype(ia)> sax(j, true);
+                json_sax_dom_parser<json, typename nlohmann::detail::container_input_adapter_factory_impl::container_input_adapter_factory<std::string>::adapter_type> sax(j, true);
```

Comment:
Well, that's a bit more ugly than I expected.  Maybe an `input_adapter_type` alias template would be useful here.

Reply:
Just to confirm, the suggestion is to add a `using string_input_adapter = decltype(input_adapter(std::declval<std::string>()));` in input_adapters.hpp or keep the `input_adapter_type` alias scoped to the two tests: unit-deserializer.cpp and unit-disabled_exceptions.cpp?

Reply:
Added